src/manifest/store.rs

```diff
@@ -23,6 +24,51 @@ use std::time::SystemTime;
 use tracing::warn;
 use uuid::Uuid;
 
+/// Represents a local view of the manifest that is in the process of being updated
+#[derive(Clone, Debug)]
+pub(crate) struct DirtyManifest {
+    id: u64,
+    next_id: u64,
+    parent: Option<ParentDb>,
+    pub(crate) core: CoreDbState,
```

Comment:
CoreDbState -> `CoreManifestState` or `ManifestState`? iiuc, it mostly reflects the state of manifest file instead of a running db instance

Reply:
I like manifest state

Reply:
I totally agree we should rename this, but I wanted to leave it out of this PR to try to keep the changes more focused.